sklearn/metrics/pairwise.py

```diff
@@ -130,6 +135,14 @@ def check_pairwise_arrays(
         .. versionchanged:: 0.23
            Accepts `pd.NA` and converts it into `np.nan`.
 
+    only_check_num_samples : bool, default=False
+        Whether to only check the length of the array.
+        When `True`, `dtype` is ignored and the check for 2D array is not performed.
```

Comment:
which check for 2D arrays is not performed? It's a bit unclear from the description what this does.